src/libs/ReportUtils.ts

```diff
@@ -1202,51 +1041,41 @@ function getDisplayNamesWithTooltips(personalDetailsList, isMultipleParticipantR
 
             // Then fallback on accountID as the final sorting criteria.
             return first.accountID > second.accountID;
-        })
-        .value();
+        });
 }
 
 /**
  * Gets a joined string of display names from the list of display name with tooltip objects.
- *
- * @param {Object} displayNamesWithTooltips
- * @returns {String}
  */
-function getDisplayNamesStringFromTooltips(displayNamesWithTooltips) {
-    return _.filter(
-        _.map(displayNamesWithTooltips, ({displayName}) => displayName),
-        (displayName) => !_.isEmpty(displayName),
-    ).join(', ');
+function getDisplayNamesStringFromTooltips(displayNamesWithTooltips: PersonalDetails[]) {
+    return displayNamesWithTooltips
+        .map(({displayName}) => displayName)
+        .filter((displayName) => !_.isEmpty(displayName))
+        .join(', ');
 }
 
 /**
  * Get the report given a reportID
- *
- * @param {String} reportID
- * @returns {Object}
  */
-function getReport(reportID) {
+function getReport(reportID: string | undefined): OnyxEntry<Report> {
     // Deleted reports are set to null and lodashGet will still return null in that case, so we need to add an extra check
-    return lodashGet(allReports, `${ONYXKEYS.COLLECTION.REPORT}${reportID}`, {}) || {};
+    return allReports?.[`${ONYXKEYS.COLLECTION.REPORT}${reportID}`] ?? ({} as OnyxEntry<Report>);
 }
 
 /**
  * Determines if a report has an IOU that is waiting for an action from the current user (either Pay or Add a credit bank account)
- *
- * @param {Object} report (chatReport or iouReport)
- * @returns {boolean}
  */
-function isWaitingForIOUActionFromCurrentUser(report) {
+function isWaitingForIOUActionFromCurrentUser(report: OnyxEntry<Report>): boolean {
     if (!report) {
         return false;
     }
 
-    if (isArchivedRoom(getReport(report.parentReportID))) {
+    if (isArchivedRoom(getReport(report.parentReportID ?? ''))) {
```

Comment:
@kubabutkiewicz Maybe this will be more clear
```suggestion
    if (report.parentReportID && isArchivedRoom(getReport(report.parentReportID))) {
```